### PR TITLE
feat: add WatsonX Responses API integration test recordings

### PR DESCRIPTION
**Pull request by leseb**
## Summary

- Fix WatsonX provider bugs: `check_model_availability` cache key mismatch, strip unsupported `parallel_tool_calls` param, handle missing `service_tier` in LiteLLM streaming chunks
- Add WatsonX provider metadata collection (build version/date from `foundation_model_specs` router_info) to the api_recorder litellm path
- Add WatsonX-specific test skips for unsupported features: tool calling, vision/image inputs, logprobs, structured output edge cases, file search filters, `max_output_tokens`
- Include 81 test recordings with provider metadata (`ibm/granite-3-8b-instruct`)

Test results: **77 passed, 129 skipped, 0 failures**

## Test plan

- [x] Run integration tests with `--setup watsonx --inference-mode record`
- [x] Verify recordings contain provider metadata (`watsonx_build`, `watsonx_build_date`)
- [x] Verify provider compatibility matrix includes WatsonX data
- [x] CI passes in replay mode

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by cdoern** [comment]
we discussed this at community call today, lets hold this for a second. I am opening a PR to fix the recording workflow, we should have that in place and decide which providers with API keys we want to support before merging more of these in

**Comment by leseb** [comment]
closing in favor of https://github.com/llamastack/llama-stack/pull/5123

**Comment by cdoern** [comment]
yeah, we might need this before we add auto-re-recording for Watsonx, because we need pre-existing recordings for the CI tests to pass.

File: src/llama_stack/providers/inline/agents/builtin/responses/streaming.py [hunk 1]
```diff
@@ -930,8 +930,10 @@ async def _process_streaming_chunks(
             chunk_created = chunk.created
             chunk_model = chunk.model
             # Extract service_tier from chunk if available (may be in final chunk)
-            if chunk.service_tier is not None:
-                chunk_service_tier = chunk.service_tier
+            # Use getattr since some providers (e.g. LiteLLM/WatsonX) may not include service_tier
+            chunk_st = getattr(chunk, "service_tier", None)
+            if chunk_st is not None:
+                chunk_service_tier = chunk_st
```
Comment (mattf) [inline src/llama_stack/providers/inline/agents/builtin/responses/streaming.py]: is this necessary after #5133 ?

Comment (leseb) [reply src/llama_stack/providers/inline/agents/builtin/responses/streaming.py]: good point, let me see check, i'm recording again

Comment (leseb) [reply src/llama_stack/providers/inline/agents/builtin/responses/streaming.py]: ok not needed, removed

**Comment by leseb** [comment]
@cdoern ok this is ready
